src/schema/v2/CollectorProfile/helpers/formatName.ts

```diff
@@ -0,0 +1,33 @@
+type NameFormatType = "default" | "public"
+
+const formatName = (name: string, format: NameFormatType): string => {
+  switch (format) {
+    case "public":
+      return formatPublicName(name)
+    case "default":
+    default:
+      return name
+  }
+}
+
+export const formatPublicName = (name: string): string => {
+  if (!name || typeof name !== "string") {
+    throw new Error("Invalid name.")
+  }
+
+  const nameParts = name.split(" ")
+
+  if (nameParts.length === 1) {
```

Comment:
for readability, it is better to avoid nesting ifs, the advantage of the functional approach is we end up with the function with the returns. Suggestion:

```jsx
if (nameParts.length === 1) {
  ...
  return nameParts[0]
}
if (nameParts.length > 1) {
  ...
  return `${firstName} ${lastNameInitial}.`.trim()
}

return name
```